### ISSUE TITLE
Comparing floating point with == or != is unsafe

### ISSUE THREAD

**Issue by piaojin** [mixpanel/mixpanel-iphone#839]
**Integration Method:** CocoaPods / Carthage / Manual
`The source code`

**Xcode Version:** 
`10.1`

**Library Version:** 

**Platform:** iOS / App Extension / watchOS / tvOS
`iOS`

**Language:** Swift / Objective-C
`Objective-C`

**Description:** 
I encountered this error when I compiled the mixpanel source code in my project.

**Error:**
`Comparing floating point with == or != is unsafe`
![89D91769-9FBF-4603-8F44-CCAACECECD54](https://user-images.githubusercontent.com/10163672/54976601-d6b00900-4fd5-11e9-9408-f328a5297024.png)

**Solution:**
Fix this issue by adding:
```
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wfloat-equal"
/* My code triggering the warnings */
#pragma clang diagnostic pop
```
**Reference:**
https://stackoverflow.com/questions/11421756/weverything-yielding-comparing-floating-point-with-or-is-unsafe

**Expected Behavior:** 
It is better to fix the issue through the authorities.

**Comment by zihejia** [comment]
Hi @piaojin , we have released v3.4.5 that fixed the issue. Thanks! https://github.com/mixpanel/mixpanel-iphone/releases/tag/v3.4.5
Closing this one out, please let me know if you're still experiencing

**Comment by piaojin** [comment]
> Hi @piaojin , we have released v3.4.5 that fixed the issue. Thanks! https://github.com/mixpanel/mixpanel-iphone/releases/tag/v3.4.5
> Closing this one out, please let me know if you're still experiencing

The issue has been fixed. Thank you.
